File: leetcode/cpp/58.length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int ans = 0;
        char prev=' ';
        for (char c: s) {
            if (c != ' ') {
                if (prev == ' ') {
                    ans = 1;
                } else {
                    ans++;
                }
            }
            prev = c;
        }
        return ans;
    }
};

File: leetcode/go/58.length-of-last-word.go
func lengthOfLastWord(s string) int {
    ans := 0
    prev := ' '
    for _, r := range s {
        if r != ' ' {
            if prev != ' ' {
                ans++
            } else {
                ans = 1
            }
        }
        prev = r
    }
    return ans
}

